modules/apps/portal-template/portal-template-xsl/src/main/java/com/liferay/portal/template/xsl/internal/XSLTemplate.java

```diff
@@ -207,6 +219,12 @@ public void prepare(HttpServletRequest request) {
 
 	@Override
 	public void processTemplate(Writer writer) throws TemplateException {
+		Thread thread = Thread.currentThread();
```

Comment:
Hi @topolik,
 
Any reason you have to do this context switching? I tested it locally, as long as the transformer is from the right jar, the transform process should be OK. I am not sure if I missed something here, please correct me  if I was wrong.

Thanks.
Tina

Reply:
Hi Tina,

when I tested it I found there are some static inits in Xalan that fires with a first template render.

We could also render a fake template to initialize them in init but this looked safer.

Reply:
Xalan was complaining that "UTF-8 is not valid Java character set and that it will use UTF-8 instead". It was org.apache.xml.serializer.Encodings#loadEncodingInfo

Reply:
Basically this: https://stackoverflow.com/questions/177242/the-encoding-utf-8-is-not-supported-by-the-java-runtime

And this answer/comment:
> The problem in my instance arose from the usage of OSGi and class-loader visibility. Multiple bundles instantiated the Xerces XML parser, but failed to find character set definitions as those resources were not visible. – Mark Derricutt Mar 2 '09 at 3:23